content/docs/nextjs/github-public-repo.md

```diff
@@ -0,0 +1,401 @@
+
+After creating the app, you should see a page with information such as **Client ID** and **Client Secret**. Next, we'll add those as environment variables to the project to connect this App to the Tina-GitHub helpers.
+
+### Setting Environment Variables
+
+[Environment variables](https://nextjs.org/docs/api-reference/next.config.js/environment-variables) are sensitive values specific to your project. The Tina-GitHub helpers will use these values to talk to your repository, enabling auth and data fetching via GitHub.
+
+To set these variables, create a `.env` file in your project root. Add the _secret_ and _id_ values from the OAuth App, and fill in the repo name. _Do not commit this file_; you may need to add `.env` to the `.gitignore` file.
+
+```
+# .env
+GITHUB_CLIENT_ID= # Taken from GitHub
+GITHUB_CLIENT_SECRET= # Taken from Github
+REPO_FULL_NAME=tinacms/tinacms.org # This is your github repository's owner / repo-name.
+BASE_BRANCH=master
+```
+
+> Make sure to **delete the comments** in this file after filling in the correct values! Otherwise the authentication may not work.
+
+You can use the `dotenv` package to load the `.env` file:
+
+```bash
+yarn add dotenv
+```
+
+Now, to load these `.env` values in the front-end, create a file called [next.config.js](https://nextjs.org/docs/api-reference/next.config.js/introduction) in the root of your project. Add the code from this example:
+
+```js
+// next.config.js
+
+require('dotenv').config()
+
+module.exports = {
+  env: {
+    GITHUB_CLIENT_ID: process.env.GITHUB_CLIENT_ID,
+    REPO_FULL_NAME: process.env.REPO_FULL_NAME,
+    BASE_BRANCH: process.env.BASE_BRANCH,
+  },
+  // ...
+}
+```
+
+> Note that we did not add `GITHUB_CLIENT_SECRET` to the config exports. This is the a secret key that should only be used from the server and should not be accessible through the browser.
+
+At this point you should be able to authenticate with GitHub! Try clicking the 'Edit This Site' button again and try to authenticate.
+
+<!-- TODO: add an image here to show auth flow modal steps? -->
+
+If all is successful, you should see another modal prompting you to create a fork, go ahead and do so. If you weren't the owner of this repository, a fork would be made where someone else could commit edits and create a PR for review. Since you're the owner of this repository, a fork won't be created and edits will just go to the `master` branch.
+
+## Loading content from GitHub
+
+Now that we have authentication set up, it's time to wire up some content to edit.
+
+Check out `pages/index.tsx`, right now the content for this page is statically written into the component. Let's create a data file to source this content from. Add a new directory in the root of your project called `content` (or `data`, whichever you prefer) and create a new file called `home.json`. We'll start small by just adding editing a title.
+
+```json
+// content/home.json
```

Comment:
This comment will break the json. Should we add a note to remove it to the comment? or maybe move it outside the codeblock?

Reply:
Yeah I think convention is usually to put the filename bolded and outside of the codeblock:

    **content/home.json**
    ```json
    {
      "title": "hello world"
    }
    ```

Then the name doesn't end up in the file when you copy-paste it to your site

Reply:
PR: https://github.com/tinacms/tinacms.org/pull/373

When this gets merged maybe I'll make a separate PR for the rest of the docs